deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,276 @@
+        dropout: float (default 0.0)
+            the dropout probablity to use.
+
+        """
+        if dropout < 0 or dropout > 1:
+            raise ValueError("dropout probability has to be between 0 and 1, "
+                             "but got {}".format(dropout))
+        model = DTNN(n_tasks=n_tasks,
+                     n_embedding=n_embedding,
+                     n_hidden=n_hidden,
+                     n_distance=n_distance,
+                     distance_min=distance_min,
+                     distance_max=distance_max,
+                     output_activation=output_activation,
+                     mode=mode,
+                     dropout=dropout)
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
+        super(DTNNModel, self).__init__(model, L2Loss(), ["prediction"],
+                                        **kwargs)
+
+    def compute_features_on_batch(self, X_b):
+        """Computes the values for different Feature Layers on given batch
+
+        A tf.py_func wrapper is written around this when creating the
+        input_fn for tf.Estimator
+
+        """
+        distance = []
+        atom_membership = []
+        distance_membership_i = []
+        distance_membership_j = []
+        num_atoms = list(map(sum, X_b.astype(bool)[:, :, 0]))
+        atom_number = [
+            np.round(
+                np.power(2 * np.diag(X_b[i, :num_atoms[i], :num_atoms[i]]),
+                         1 / 2.4)).astype(int) for i in range(len(num_atoms))
+        ]
+        start = 0
+        for im, molecule in enumerate(atom_number):
+            distance_matrix = np.outer(
+                molecule, molecule) / X_b[im, :num_atoms[im], :num_atoms[im]]
+            np.fill_diagonal(distance_matrix, -100)
+            distance.append(np.expand_dims(distance_matrix.flatten(), 1))
+            atom_membership.append([im] * num_atoms[im])
+            membership = np.array([np.arange(num_atoms[im])] * num_atoms[im])
+            membership_i = membership.flatten(order='F')
+            membership_j = membership.flatten()
+            distance_membership_i.append(membership_i + start)
+            distance_membership_j.append(membership_j + start)
+            start = start + num_atoms[im]
+
+        atom_number = np.concatenate(atom_number).astype(np.int32)
+        distance = np.concatenate(distance, axis=0)
+        gaussian_dist = np.exp(-np.square(distance - self.model.steps) /
+                               (2 * self.model.step_size**2))
+        gaussian_dist = gaussian_dist.astype(np.float64)
+        atom_mem = np.concatenate(atom_membership).astype(np.int64)
+        dist_mem_i = np.concatenate(distance_membership_i).astype(np.int64)
+        dist_mem_j = np.concatenate(distance_membership_j).astype(np.int64)
+
+        features = [
+            atom_number, gaussian_dist, atom_mem, dist_mem_i, dist_mem_j
+        ]
```

Comment:
These features should be documented. What do they all mean? Potentially this method may be useful as a standalone featurizer.

Reply:
Done

Reply:
This code is still quite complex. We need more comments through the code and possibly need to factor out some helper functions to simplify